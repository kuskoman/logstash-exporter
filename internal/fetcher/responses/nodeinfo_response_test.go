package responses_test

import (
	"encoding/json"
	"os"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"

	"github.com/kuskoman/logstash-exporter/internal/fetcher/responses"
)

func TestNodeInfoResponseStructure(t *testing.T) {
	fixtureContent, err := os.ReadFile("../../../fixtures/node_info.json")
	if err != nil {
		t.Fatalf("Error reading fixture file: %v", err)
	}

	var target responses.NodeInfoResponse
	err = json.Unmarshal(fixtureContent, &target)
	if err != nil {
		t.Fatalf("Error unmarshalling fixture: %v", err)
	}

	snaps.MatchSnapshot(t, "Unmarshalled NodeInfoResponse", target)
}
