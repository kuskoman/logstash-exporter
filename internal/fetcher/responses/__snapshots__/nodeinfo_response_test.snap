
[TestNodeInfoResponseStructure - 1]
Unmarshalled NodeInfoResponse
responses.NodeInfoResponse{
    Host:          "9e6e14cf5532",
    Version:       "8.6.1",
    HTTPAddress:   "0.0.0.0:9600",
    ID:            "a2c7110e-5ccf-4226-bc9b-e773710e66a0",
    Name:          "9e6e14cf5532",
    EphemeralID:   "4a2ee04f-2733-4eaa-887d-675bad27f07c",
    Status:        "green",
    Snapshot:      false,
    Pipeline:      struct { Workers int "json:\"workers\""; BatchSize int "json:\"batch_size\""; BatchDelay int "json:\"batch_delay\"" }{Workers:16, BatchSize:125, BatchDelay:50},
    BuildDate:     "2023-01-24T10:41:55+00:00",
    BuildSHA:      "6a248b5091c490f09460db5651e5239d903b97cf",
    BuildSnapshot: false,
}
---
